patches/node/fix_key_gen_apis_are_not_available_in_boringssl.patch

```diff
@@ -124,3 +72,54 @@ index 6473b652ac95609aff555d99be38b48a5aa513a5..caaaf19dc02101c2024b511780c94fc8
      buf += 2;
      Local<Object> obj = Object::New(env->isolate());
      if (!Set(env->context(),
+diff --git a/src/crypto/crypto_dsa.cc b/src/crypto/crypto_dsa.cc
+index 271db427fa8539feb30c1712574976fb1f623e91..05ec6103007b0c687c1c713c44f5e00f1f2f33c3 100644
+--- a/src/crypto/crypto_dsa.cc
++++ b/src/crypto/crypto_dsa.cc
+@@ -29,7 +29,7 @@ namespace crypto {
+ EVPKeyCtxPointer DsaKeyGenTraits::Setup(DsaKeyPairGenConfig* params) {
+   EVPKeyCtxPointer param_ctx(EVP_PKEY_CTX_new_id(EVP_PKEY_DSA, nullptr));
+   EVP_PKEY* raw_params = nullptr;
+-
++#ifndef OPENSSL_IS_BORINGSSL
+   if (!param_ctx ||
+       EVP_PKEY_paramgen_init(param_ctx.get()) <= 0 ||
+       EVP_PKEY_CTX_set_dsa_paramgen_bits(
+@@ -49,7 +49,7 @@ EVPKeyCtxPointer DsaKeyGenTraits::Setup(DsaKeyPairGenConfig* params) {
+       return EVPKeyCtxPointer();
+     }
+   }
+-
++#endif
```

Comment:
are we correctly signalling failure here?

Reply:
probably not but we historically haven't either and i'm not sure how we can do that in this context since nothing in that block is exposed via BoringSSL 🤔